### PR TITLE
docs: use nixosOptionDocs

### PR DESCRIPTION
**Pull request by NotAShelf**
NixOS 23.11 is deprecating DocBook option documentation. Following home-manager in this change is probably a good idea. This change drops `literalDocBook` and the provided nmd functions in favor of nixosOptionDocs and CommonMark in option descriptions.

### CODE REVIEW

**Comment by NotAShelf** [comment]
@horriblename building options.json currently fails with `option vim.extraPlugins.<name>.package has no description`, any chance you could look at this?

**Comment by horriblename** [comment]
oops missing description

```diff
diff --git a/lib/types/plugins.nix b/lib/types/plugins.nix
index 344c405..ee33a42 100644
--- a/lib/types/plugins.nix
+++ b/lib/types/plugins.nix
@@ -112,6 +112,7 @@ with lib; let
       options = {
         package = mkOption {
           type = pluginType;
+          description = "Plugin Package.";
         };
         after = mkOption {
           type = listOf str;
```

build still not passing tho, this time failing on

```
       > /nix/store/q53bq2b1dw0wa75znak2sxi1p5rxkl5g-manual-combined/manual-combined.xml:7
45: element appendix: Relax-NG validity error : Element appendix failed to validate conten
t
       >    741   </section>
       >    742      </chapter>
       >    743       <appendix xml:id="ch-options">
       >    744   <title>Configuration Options</title>
       >    745    <appendix xmlns="http://docbook.org/ns/docbook" xml:id="appendix-config
uration-options" xml:base="nmd-result/neovim-flake-options.xml">
       >    746         <title>Configuration Options</title>
       >    747  <variablelist xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:nixos="tag
:nixos.org" xmlns="http://docbook.org/ns/docbook" xml:id="neovim-flake-options">
       >
       > /nix/store/q53bq2b1dw0wa75znak2sxi1p5rxkl5g-manual-combined/manual-combined.xml f
ails to validate

```

do you know what's going on?

**Comment by NotAShelf** [comment]
I might have an idea, I'll see if that helps

**Comment by NotAShelf** [comment]
neat, that did the trick

**Comment by horriblename** [comment]
`{option}` don't provide links in the html docs.

i.e.

> If `vim.lsp.lspkind.enable` is true, then the function will be called before modifications from lspkind.

instead of

> <p class="simpara">If <a class="xref" href="options.html#opt-vim.lsp.lspkind.enable"><code class="option">vim.lsp.lspkind.enable</code></a> is true, then the function will be called before modifications from lspkind.</p>

**Comment by NotAShelf** [comment]
that is interesting, we're basically following home-manager in regards of nmd usage - but features that it's able to use don't apply to us, which seems to be all CommonMark features?

**Comment by NotAShelf** [comment]
after little research, I've noticed that `{option}` *intentionally* does not link the relevant issue. Home-manager seems to be using `[](#opt-home...)` for linked options.

**Comment by NotAShelf** [comment]
Okay, I believe [`9328443`](https://github.com/NotAShelf/neovim-flake/pull/177/commits/93284432739b79425638d7e707252e24c7cbe5dc) remedies the issue you've mentioned. The only remaining issue is the codeblock not being converted correctly in `autocomplete.formatting.format`

image

**Comment by NotAShelf** [comment]
Might be because it's an example?

**Comment by horriblename** [comment]
`lib.literalMD` seems to work

File: modules/completion/nvim-cmp/nvim-cmp.nix [hunk 1]
```diff
@@ -40,23 +40,22 @@ with builtins; {
...
             will be called before modifications from lspkind.
 
             Default is to call the menu mapping function.
           '';
           type = types.str;
           default = "nvim_cmp_menu_map";
-          example = nvim.nmd.literalAsciiDoc ''
-            [source,lua]
-            ---
+          example = ''
```
Comment (horriblename) [inline modules/completion/nvim-cmp/nvim-cmp.nix:53]:
```suggestion
          example = lib.literalMD ''
```

**Review by horriblename — DISMISSED** [review]
LGTM

**Pull request merged**